tests/main/interfaces-firewall-control/task.yaml

```diff
@@ -23,32 +23,37 @@ environment:
     DESTINATION_IP: "172.26.0.15"
 
 prepare: |
+    #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB/snaps.sh"
     echo "Given a snap declaring a plug on the firewall-control interface is installed"
     install_local firewall-control-consumer
 
     echo "And a service is listening"
-    printf "#!/bin/sh -e\nwhile true; do echo \"HTTP/1.1 200 OK\n\nok\n\" |  nc -l -p $PORT -w 1; done" > $SERVICE_FILE
-    chmod a+x $SERVICE_FILE
+    printf '#!/bin/sh -e\nwhile true; do echo "HTTP/1.1 200 OK\n\nok\n" |  nc -l -p %s -w 1; done' "$PORT" > "$SERVICE_FILE"
+    chmod a+x "$SERVICE_FILE"
+    #shellcheck source=tests/lib/systemd.sh
     . "$TESTSLIB/systemd.sh"
-    systemd_create_and_start_unit $SERVICE_NAME "$(readlink -f $SERVICE_FILE)"
+    #shellcheck disable=SC2153
+    systemd_create_and_start_unit "$SERVICE_NAME" "$(readlink -f "$SERVICE_FILE")"
 
-    while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\n*"; do sleep 0.5; done
+    while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\\n*"; do sleep 0.5; done
 
     echo "And we store a basic HTTP request"
-    cat > $REQUEST_FILE <<EOF
+    cat > "$REQUEST_FILE" <<EOF
     GET / HTTP/1.0
 
     EOF
 
 restore: |
+    #shellcheck source=tests/lib/systemd.sh
     . "$TESTSLIB/systemd.sh"
-    systemd_stop_and_destroy_unit $SERVICE_NAME
-    rm -f firewall-create.error $SERVICE_FILE $REQUEST_FILE
+    #shellcheck disable=SC2153
+    systemd_stop_and_destroy_unit "$SERVICE_NAME"
+    rm -f firewall-create.error "$SERVICE_FILE" "$REQUEST_FILE"
 
 execute: |
     echo "Then the plug is disconnected by default"
-    snap interfaces -i firewall-control | MATCH "^\- +firewall-control-consumer:firewall-control"
+    snap interfaces -i firewall-control | MATCH '^\- +firewall-control-consumer:firewall-control'
```

Comment:
why is the `-` escaped? it's not a metacharacter

Reply:
(note there's a bunch of these)

Reply:
```
+ MATCH '- +test-snapd-kernel-module-consumer:kernel-module-control'
+ snap interfaces -i kernel-module-control
grep: invalid option -- ' '
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
```

Hmm, so either current escaping approach or using `MATCH -- <pattern>`.

Reply:
+1 for --

Reply:
Hmm found a nicer way to do this `MATCH '^- +test-snapd-kernel-module-consumer:kernel-module-control'` that still makes grep happy.